### PR TITLE
Integrate Java Maven tests into cephCI

### PR DESCRIPTION
**Pull request by TejasC88**
Integrate Java Maven suite into cephCI

Pass log : http://magna002.ceph.redhat.com/ceph-qe-logs/tejas/cephci-run-NGU4NF/

### CODE REVIEW

File: tests/rgw/java_s3tests.py [hunk 1]
```diff
@@ -51,14 +51,15 @@ def run(**kw):
...
-
     if not host:
-        _, secure, _ = get_rgw_frontend(cluster)
+        _, secure, port = get_rgw_frontend(cluster)
 
-    exit_status = execute_s3_tests(client_node, config)
-    execute_teardown(cluster)
+    if "maven" in config["suite"]:
+        exit_status = maven_setup(cluster, config, port)
+    elif "JavaS3" in config["suite"]:
```
Comment (mkasturi18) [inline tests/rgw/java_s3tests.py:59]: @TejasC88 , can we use if statement here as well.

File: tests/rgw/java_s3tests.py [hunk 2]
```diff
@@ -289,3 +290,64 @@ def create_s3_user(node: CephNode, data: Dict) -> None:
...
+    repo_url = "https://github.com/ceph/ceph.git"
+    node.exec_command(cmd=f"git clone {repo_url}")
+    node.exec_command(cmd="yum install java-17-openjdk -y", sudo=True)
+    node.exec_command(cmd="alternatives --config java <<< '2'", sudo=True)
+    rgw_endpoint = data["endpoint"]
+    access_key = data["user1"]["access_key"]
+    secret = data["user1"]["secret_key"]
+
+    log.info("Executing Java Maven test suite")
+    # node.exec_command(cmd=f"cd ceph/qa/workunits/rgw/jcksum/")
```
Comment (mkasturi18) [inline tests/rgw/java_s3tests.py:339]: remove line with comment

File: suites/reef/rgw/tier-2_ssl_rgw_ecpool_test.yaml [hunk 3]
```diff
@@ -166,6 +168,18 @@ tests:
       name: execute Java s3tests
       polarion-id: CEPH-83586289
 
+  - test:
+      abort-on-fail: false
+      config:
+        suite:
+          - maven
+        ssl: true
```
Comment (mkasturi18) [inline suites/reef/rgw/tier-2_ssl_rgw_ecpool_test.yaml:176]: @TejasC88 , can you include this test in non-ssl suite as well for better coverage

**Review by mkasturi18 — CHANGES_REQUESTED** [review]

File: tests/rgw/java_s3tests.py (hunk 1 above)
Comment (TejasC88) [reply tests/rgw/java_s3tests.py]: Done

File: tests/rgw/java_s3tests.py (hunk 2 above)
Comment (TejasC88) [reply tests/rgw/java_s3tests.py]: Done

File: suites/reef/rgw/tier-2_ssl_rgw_ecpool_test.yaml (hunk 3 above)
Comment (TejasC88) [reply suites/reef/rgw/tier-2_ssl_rgw_ecpool_test.yaml:176]: @mkasturi18 added to the regression extended suite as well 
pass log : magna002.ceph.redhat.com:/ceph/ceph-qe-logs/tejas/cephci-run-SUS31D

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**